foodnet/membership/models.py

```diff
@@ -53,6 +53,18 @@ def is_complete(self):
             return True
         return False
 
+    def has_admin_permission(self, department):
+        return self.departmentadministrator_set.filter(department=department).exists()
+
+    def can_be_edited_by(self, user_profile):
+        # can be edited by user_profile if the user_profile is an admin of
+        # the self.member.department department
+        departments = self.member.department_set.all()
+        for department in departments.iterator():
+            if user_profile.has_admin_permission(department=department):
```

Comment:
I hate leaky abstractions :)
I suppose that if the need arise, we can substitute with some good ol' piece of SQL.